unit_test/sparse/Test_Sparse_spmv.hpp

```diff
@@ -208,11 +233,19 @@ void check_spmv_mv(crsMat_t input_mat, x_vector_type x, y_vector_type y, y_vecto
                             typename y_vector_type::non_const_value_type beta,
                             int numMV) {
     typedef typename crsMat_t::execution_space ExecSpace;
-    typedef typename crsMat_t::values_type::non_const_type scalar_view_t;
-    typedef typename scalar_view_t::value_type ScalarA;
     typedef Kokkos::RangePolicy<ExecSpace> my_exec_space;
 
-    double eps = std::is_same<ScalarA,float>::value?2*1e-3:1e-7;
+    // y is the quantity being tested here,
+    // so let us use y_value_type to determine
+    // the appropriate tolerance precision.
+    using y_value_type  = typename y_vector_type::non_const_value_type;
+    using y_value_trait = Kokkos::ArithTraits<y_value_type>;
+    // Note on 2019-12-03 by LBV:
+    // one really want the following definition,
+    // const y_value_trait::mag_type eps = f(y_value_trait::eps());
+    // where f() introduces a fudge factor that accounts for
+    // rounding error and such...
+    const double eps = std::is_same<y_value_trait::mag_type, float>::value ? 2*1e-3 : 1e-7;
```

Comment:
`typename` is needed in `std::is_same` to remove compilation error - `typename y_value_trait::mag_type`